tests/TestModified.cpp

```diff
@@ -469,19 +471,21 @@ void TestModified::testHistoryItem()
     entry4->setGroup(root);
     QCOMPARE(entry4->historyItems().size(), 0);
 
+    const QString key("test");
+
     int reservedSize = entry4->attributes()->attributesSize();
     entry4->beginUpdate();
-    entry4->attachments()->set("test1", QByteArray(17000 - 5 - reservedSize + 1, 'a'));
+    entry4->attachments()->set(key, QByteArray(historyMaxSize - key.size() - reservedSize + 1, 'a'));
     entry4->endUpdate();
     QCOMPARE(entry4->historyItems().size(), 1);
```

Comment:
I'm still a little confused here. Aren't you just repeating the above tests with different attachments? I think you actually want to test historySize, not historyItems.

Reply:
The tests ensure that all types of associations, attachments, attributes and auto-types (which were not stored to the history at all) with their corresponding keys are considered for the size calculation and prevents regressions in this regard. Maybe the cases should be moved to a separate test case, but the changes were to little to justify this in my eyes.

Reply:
And that's good. I agree they are not entirely redundant. However, I don't see you testing that an entry with an attachment of n bytes actually reports back n + x bytes as size or that adding another attachment/auto-type association removes the oldest history item. It would be amazing if you could add a few quick tests for that as well.